packages/create-toolpad-app/index.ts

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env node
+
+import chalk from 'chalk';
+import { execaCommand } from 'execa';
+import inquirer from 'inquirer';
+import * as fs from 'fs/promises';
+import path from 'path';
+import { getPackageManager } from './utils/getPackageManager';
+import { isFolderEmpty } from './utils/isFolderEmpty';
+// Detect the package manager
+const packageManager = getPackageManager();
+
+// Install the dependencies
+const installDeps = async () => {
+  // eslint-disable-next-line no-console
+  console.log(`${chalk.blue('info')} - Installing dependencies`);
+
+  const installVerb = packageManager === 'yarn' ? 'add' : 'install';
+  const command = `${packageManager} ${installVerb} @mui/toolpad @mui/toolpad-core`;
+  await execaCommand(command, { stdio: 'inherit' });
+
+  // eslint-disable-next-line no-console
+  console.log(`${chalk.green('success')} - Dependencies installed successfully!`);
+};
+
+// Create a new directory and initialize a new project
+const scaffoldProject = async () => {
+  const name = await inquirer.prompt([
+    {
+      type: 'input',
+      name: 'projectName',
+      message: 'Enter the name of your project:',
+      default: 'my-toolpad-app',
+    },
+  ]);
+
+  // eslint-disable-next-line no-console
+  console.log(`Creating a new MUI Toolpad project in ${chalk.blue(name.projectName)}`);
+  try {
+    await fs.mkdir(name.projectName);
+    process.chdir(name.projectName);
+
+    const packageJson = {
+      name: name.projectName,
+      version: '0.1.0',
+      private: true,
+      scripts: {
+        dev: 'toolpad dev',
+        build: 'toolpad build',
+        start: 'toolpad start',
+      },
+    };
+
+    await fs.writeFile('package.json', JSON.stringify(packageJson, null, 2));
+    return name.projectName;
+  } catch (error) {
+    // Directory exists, verify if it is empty to continue
+    if (!isFolderEmpty(path.join(process.cwd(), name.projectName), name.projectName)) {
```

Comment:
Ideally, you read `process.cwd()` once and pass it around. Especially when you'll be doing `process.chdir` in some places.
The Purer a functions is, the easier to debug and test.